### PR TITLE
add a test for filtering by profile status

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description
I wasn't clear on whether or not status filtering on compliance reporting profiles is supported or not, so i added a test. And it is!! 🎉 
Figured we should commit the test, so we know for sure that it's supported

### :+1: Definition of Done
test for filtering by status on compliance reporting profiles

### Related Resources
https://github.com/chef/automate/issues/443

### CODE REVIEW

**Comment by vjeffrey** [comment]
in conversations with @rickmarry, he stated this may be reflecting the overall report status and not the profile status. i'm going to look into this locally.

**Comment by vjeffrey** [comment]
i've confirmed it is doing the profile status.
basically i did a scan job with 3 profiles. 2 failed, one skipped. overall status is obviously failed, so i tried to pull the compliance/profiles endpoint with a filter of status: skipped and it only returned the skipped profile

File: components/compliance-service/api/tests/08_search_profiles_spec.rb [hunk 1]
```diff
@@ -180,6 +180,30 @@ def reporting;
...
+                "version" => "2.1.0",
+                "status" => "passed"
+            }
+        ],
+        "counts" => {
+            "total" => 1,
+            "passed" => 1
+        }
+    }.to_json
+    assert_equal(expected_data, actual_data.to_json)
```
Comment (phiggins) [inline components/compliance-service/api/tests/08_search_profiles_spec.rb]: We have a helper function for this kind of test: https://github.com/chef/automate/blob/53665d66aaeb9c4cebb1d54ce41fb16375665ae0/components/compliance-service/api/tests/run.rb#L105-L107

**Comment by phiggins** [comment]
> basically i did a scan job with 3 profiles. 2 failed, one skipped. overall status is obviously failed, so i tried to pull the compliance/profiles endpoint with a filter of status: skipped and it only returned the skipped profile

Is there a way to test this scenario in code? Probably not worth redoing the ruby integration test data for this but it seems worth testing if we can.

File: components/compliance-service/api/tests/08_search_profiles_spec.rb (hunk 1 above)
Comment (vjeffrey) [reply components/compliance-service/api/tests/08_search_profiles_spec.rb]: ohh nice!

**Comment by vjeffrey** [comment]
> Is there a way to test this scenario in code? 

i'm gonna parse through our data we send in and see if i can find a good example. if we run a skipped profile on a failed node and then try to filter for the skipped profile that should give us what we want

**Pull request merged**